apps/schedule/feeds.py

```diff
@@ -32,26 +35,150 @@ def _format_event_description(event):
 
 
 @schedule.route("/schedule/<int:year>.json")
-@cross_origin(methods=["GET"])
 def schedule_json(year):
     if year != event_year():
         return feed_historic(year, "json")
 
-    if not feature_enabled('SCHEDULE'):
+    if not feature_enabled("SCHEDULE"):
         abort(404)
 
-    schedule = [_convert_time_to_str(p) for p in _get_scheduled_proposals(request.args)]
+    if request.args.get('format') == 'array':
+        schedule = [_convert_time_to_str(p) for p in _get_scheduled_proposals(request.args)]
 
-    # NB this is JSON in a top-level array (security issue for low-end browsers)
-    return Response(json.dumps(schedule), mimetype="application/json")
+        # NB this is JSON in a top-level array (security issue for low-end browsers)
+        return Response(json.dumps(schedule), mimetype="application/json")
+
+    def duration_hhmm(duration_minutes):
+        if not duration_minutes or duration_minutes < 1:
+            return "00:00"
+        return "{}:{}".format(
+            int(duration_minutes / 60),
+            str(duration_minutes % 60).zfill(2),
+        )
+
+    schedule = (
+        Proposal.query.filter(
+            Proposal.is_accepted,
+            Proposal.scheduled_time.isnot(None),
+            Proposal.scheduled_venue_id.isnot(None),
+            Proposal.scheduled_duration.isnot(None),
+        )
+        .order_by(Proposal.scheduled_time)
+        .all()
+    )
+
+    duration_days = ceil((event_end() - event_end()).total_seconds / 86400)
+
+    rooms = [proposal.scheduled_venue.name for proposal in schedule]
+
+    schedule_json = {
+        "version": "1.0-public",
+        "conference": {
+            "acronym": "emf{}".format(event_year()),
+            "days": [],
+            "daysCount": duration_days,
+            "end": event_end().strftime("%Y-%m-%d"),
+            "rooms": [
+                {
+                    "name": room,
+                }
+                for room in rooms
+            ],
+            "start": event_start().strftime("%Y-%m-%d"),
+            "time_zone_name": event_tz,
+            "timeslot_duration": "00:10",
+            "title": "Electromagnetic Field {}".format(event_year()),
+            "url": external_url("main"),
+        },
+    }
+
+    for day in range(0, duration_days):
+        day_dt = event_start() + timedelta(days=day)
+        day_schedule = {
+            "date": day_dt.strftime("%Y-%m-%d"),
+            "day_end": (day_dt.replace(hour=3, minute=59, second=59) + timedelta(days=1)).isoformat(),
+            "day_start": day_dt.replace(hour=4, minute=0, second=0).isoformat(),
+            "index": day,
+            "rooms": {},
+        }
+        for room in rooms:
+            day_schedule["rooms"][room] = []
+            for proposal in schedule:
+                if proposal.scheduled_venue.name != room:
+                    # TODO find a better way to do that
+                    continue
```

Comment:
You can do this with `order_by(Proposal.scheduled_venue_id, Proposal.scheduled_time)` and then `itertools.groupby`.